src/app/payments/helpers.ts

```diff
@@ -145,20 +152,27 @@ export const constructPaymentFlowBuilder = async ({
     usdFromBtcMidPriceFn,
     btcFromUsdMidPriceFn,
   })
-  const builderWithInvoice = uncheckedAmount
-    ? paymentBuilder.withNoAmountInvoice({ invoice, uncheckedAmount })
-    : paymentBuilder.withInvoice(invoice)
+  const builderWithInvoice = invoice
+    ? uncheckedAmount
+      ? paymentBuilder.withNoAmountInvoice({ invoice, uncheckedAmount })
+      : paymentBuilder.withInvoice(invoice)
+    : paymentBuilder.withoutInvoice({ uncheckedAmount, description: memoPayer })
 
   const builderWithSenderWallet = builderWithInvoice.withSenderWallet(senderWallet)
 
   let builderAfterRecipientStep:
     | LPFBWithRecipientWallet<WalletCurrency, WalletCurrency>
     | LPFBWithError
   if (builderWithSenderWallet.isIntraLedger()) {
-    const recipientDetails = await recipientDetailsFromInvoice(invoice)
-    if (recipientDetails instanceof Error) return recipientDetails
-    builderAfterRecipientStep =
-      builderWithSenderWallet.withRecipientWallet(recipientDetails)
+    const recipientDetailsForBuilder =
+      invoice === undefined
```

Comment:
Again don't like this logic - I think the intra-ledger and non-intra-ledger use case should perhaps not be merged as they can be clearly distinguished higher in the call stack.